test/install_test.go

```diff
@@ -299,48 +305,88 @@ func TestDashboard(t *testing.T) {
 }
 
 func TestInject(t *testing.T) {
-	var out string
-	var err error
-
-	prefixedNs := TestHelper.GetTestNamespace("smoke-test")
-
-	out, err = testutil.ReadFile("testdata/smoke_test.yaml")
+	resources, err := testutil.ReadFile("testdata/smoke_test.yaml")
 	if err != nil {
 		t.Fatalf("failed to read smoke test file: %s", err)
 	}
-	err = TestHelper.CreateNamespaceIfNotExists(prefixedNs, map[string]string{
-		k8s.ProxyInjectAnnotation: k8s.ProxyInjectEnabled,
-	})
-	if err != nil {
-		t.Fatalf("failed to create %s namespace: %s", prefixedNs, err)
-	}
 
-	out, err = TestHelper.KubectlApply(out, prefixedNs)
-	if err != nil {
-		t.Fatalf("kubectl apply command failed\n%s", out)
-	}
+	testCases := []struct {
+		ns          string
+		autoinject  bool
+		annotations map[string]string
```

Comment:
i can think of 3 ways to inject:
1. set auto-inject on namespace ✔️
2. `linkerd inject --manual` ✔️
3. regular `linkerd inject` on a non-auto-injected namespace (maybe add this one?)